src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -22,16 +24,24 @@ class LeonardoService(gdDAO: DataprocDAO, dbRef: DbReference)(implicit val execu
   }
 
   def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest): Future[Cluster] = {
-    gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
+    def create = gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
       dbRef.inTransaction { dataAccess =>
         dataAccess.clusterQuery.save(Cluster(clusterRequest, clusterResponse))
       }
     }
+
+    dbRef.inTransaction { dataAccess =>
+      dataAccess.clusterQuery.getByName(googleProject, clusterName)} flatMap {
```

Comment:
Move the `flatMap` outside of `inTransaction` so we don't have DAO code inside a DB transaction

Reply:
+1

Reply:
it's already outside of the transaction ;)

Reply:
but that was unclear, so put a newline before the close brace on line 34 so it's more obvious to other people reading the code ;)

Reply:
such sass
agreed!